healthCheck/health-check.js

```diff
@@ -1,99 +1,99 @@
-const kafka = require("./kafka");
-const { v1 : uuidv1 } = require('uuid');
-const assessmentHealthCheck = require("./assessments");
-const kendraHealthCheck = require("./kendra");
+const { healthCheckHandler } = require('elevate-services-health-check')
+const healthCheckConfig = require('./health.config')
+const { v1: uuidv1 } = require('uuid')
 
-const obj = {
-    MONGO_DB: {
-        NAME: 'Mongo.db',
-        FAILED_CODE: 'MONGODB_HEALTH_FAILED',
-        FAILED_MESSAGE: 'Mongo db is not connected'
-    },
-    KAFKA: {
-        NAME: 'kafka',
-        FAILED_CODE: 'KAFKA_HEALTH_FAILED',
-        FAILED_MESSAGE: 'Kafka is not connected'
-    },
-    ASSESSMENT_SERVICE: {
-      NAME: 'assessmentservice.api',
-      FAILED_CODE: 'ASSESSMENT_SERVICE_HEALTH_FAILED',
-      FAILED_MESSAGE: 'ASSessment service is not healthy'
-    },
-    KENDRA_SERVICE: {
-        NAME: 'kendraservice.api',
-        FAILED_CODE: 'KENDRA_SERVICE_HEALTH_FAILED',
-        FAILED_MESSAGE: 'Kendra service is not healthy'
-    },
-    NAME: 'ImprovementServiceHealthCheck',
-    API_VERSION: '1.0'
+let health_check = async function (req, res) {
+	try {
+		validateHealthConfig(healthCheckConfig)
+		const response = await healthCheckHandler(healthCheckConfig, req.query.basicCheck, req.query.serviceName)
+		res.status(200).json(response)
+	} catch (err) {
+		console.error('Health config validation failed:', err.message || err)
+		res.status(400).json({
+			id: 'mentoringService.Health.API',
+			ver: '1.0',
+			ts: new Date(),
+			params: {
+				resmsgid: uuidv1(),
+				msgid: req.headers['msgid'] || req.headers.msgid || uuidv1(),
+				status: 'failed',
+				err: 'CONFIG_VALIDATION_ERROR',
+				errMsg: err.message || 'Invalid config',
+			},
+			status: 400,
+			result: {},
+		})
+	}
 }
 
-let health_check = async function(req,res) {
+let healthCheckStatus = function (req, res) {
+	let responseData = response(req)
+	res.status(200).json(responseData)
+}
 
-    let checks = [];
-    let mongodbConnection = await mongodb.health_check();
-    let kafkaConnection = await kafka.health_check();
-    let assessmentServiceStatus = await assessmentHealthCheck.health_check();
-    let kendraServiceStatus = await kendraHealthCheck.health_check();
-    checks.push(singleCheckObj("KAFKA",kafkaConnection));
-    checks.push(singleCheckObj("MONGO_DB",mongodbConnection));
-    checks.push(singleCheckObj("ASSESSMENT_SERVICE",assessmentServiceStatus));
-    checks.push(singleCheckObj("KENDRA_SERVICE",kendraServiceStatus));
+let response = function (req, result) {
+	return {
+		id: 'User.service.Health.API',
+		ver: '1.0',
+		ts: new Date(),
+		params: {
+			resmsgid: uuidv1(),
+			msgid: req.headers['msgid'] || req.headers.msgid || uuidv1(),
+			status: 'successful',
+			err: 'null',
+			errMsg: 'null',
+		},
+		status: 200,
+		result: result,
+	}
+}
 
-    let checkServices = checks.filter( check => check.healthy === false);
+function validateHealthConfig(config) {
```

Comment:
@MallanagoudaB Please move the validation to npm package. and it should be generic enough